### PR TITLE
Wifi now component

### PR DESCRIPTION
**Pull request by motwok**
## Description:
wifi_now esphome component - communicate directly to other ESP device via ESP-Now

**Related issue (if applicable):** fixes <link to issue>
[ESP-Now integration #275](https://github.com/esphome/feature-requests/issues/275)

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#1290

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: index.rst [hunk 1]
```diff
@@ -354,6 +354,7 @@ Misc Components
     Exposure Notifications, components/exposure_notifications, exposure_notifications.png
     RTTTL Buzzer, components/rtttl, buzzer.jpg
     Prometheus, components/prometheus, prometheus.svg
+    Wifi Now, component/wifi_now, wifi_now.svg
```
Comment (jesserockz) [inline index.rst]:
```suggestion
    Wifi Now, components/wifi_now, wifi_now.svg
```

Comment (motwok) [reply index.rst]: thanks

**Comment by motwok** [comment]
Push for first annual

**Comment by nagyrobi** [comment]
Closing as the corresponding PR is outdated by the author: https://github.com/esphome/esphome/pull/1290#issuecomment-1049138410

**Pull request closed without merging**